packages/cms-drupal/src/Resolver/ChannelResolver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\ekino_rendr\Resolver;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\ekino_rendr\Entity\ChannelInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+class ChannelResolver implements ArgumentValueResolverInterface
+{
+    protected $entityTypeManager;
+
+    public function __construct(
+        EntityTypeManagerInterface $entity_type_manager
+    ) {
+        $this->entityTypeManager = $entity_type_manager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadata $argument)
+    {
+        // This might return no channel so argument must be nullable
+        return ChannelInterface::class === $argument->getType() && $argument->isNullable();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve(Request $request, ArgumentMetadata $argument)
+    {
+        $host = $request->getHost();
+        $locale = $request->getLocale();
+        $rendrChannel = $request->get('rendr_channel');
+
+        if ($rendrChannel) {
+            $rendrChannel = 0 === \strpos($rendrChannel, 'http') ? $rendrChannel : \sprintf('http://%s', $rendrChannel);
```

Comment:
I don't know much about the project so quick question out of curiosity: why default scheme is `http` and not `https`?

Reply:
this is not an exposed url. it is used to work with `parse_url`.
if not prefixed with http:// or https://, `parse_url('www.example.com')` yield unexpected results